webapp/src/components/move_thread_modal/index.ts

```diff
@@ -42,9 +44,15 @@ function mapStateToProps(state: GlobalState) {
         return myTeams;
     };
 
+    const getChannelsForTeamFunc = (teamID: string) => {
+        const channels = Client4.getChannels(teamID, 0, General.CHANNELS_CHUNK_SIZE);
+        return channels;
+    };
```

Comment:
This is ok but a bit of an anti-pattern. I'd suggest just importing the `getChannels` action from mattermost-redux and passing that in through `mapDispatchToProps` like the other actions are. That should still return you the channels you need but will also update the redux store

Reply:
Yeah, I agree that it is a bit of an anti-pattern. Using the client directly was actually recommended by Harrison for the very reason of not adding the extra redux processing from the normal action.

What do you think? Either option sounds reasonable to me, but I definitely have less of a grasp of what sort of overhead needlessly updating the team objects in the store would cause.

Reply:
I'm fine with it 👍 Some times using anti-patterns make sense

Reply:
Awesome. Thanks for the input.